libr/core/canal.c

```diff
@@ -3901,6 +3901,11 @@ static void getpcfromstack(RCore *core, RAnalEsil *esil) {
 	free (buf);
 }
 
+static inline bool canal_isThumb(RCore *core) {
```

Comment:
i would get rid of this conditional, it makes no sense to do something special with syscalls jus because its thumb or not

Reply:
We very specifically have to do different things with arm syscalls, we need to lookup the op.val as the syscall number instead of the rax or whatever register...

Reply:
I think you're claiming we need to look up the 'interrupt' (ie op.val) on all all platforms, but thats not the case

```
macos
0x01057e0b			0f05			syscall

arm thumb
0x0003e288			a9df    		svc 0xa9
or
0x0003e288			79df    		svc 0x79
etc

linux
0x3c001065			cd80			int 0x80

```

macos doesnt use any immediate at all, and again, the immediate for thumb isnt an interrupt its a svc call. So even if we were to just use op.val always, we would have to have an isThumb call to put the op.val into serviceNumber instead of interrupt